simplq/src/components/pages/Status/QueueDetails.jsx

```diff
@@ -1,9 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
+import moment from 'moment';
+import { CircularProgress } from '@material-ui/core';
 import Header from '../../common/Header';
 import styles from '../../../styles/statusPage.module.scss';
+import * as QueueService from '../../../services/queue';
+import { handleApiErrors } from '../../ErrorHandler';
 
 export default (props) => {
   // https://dabblet.com/gist/1506530 --> checkbox hack
+  const [details, setDetails] = useState();
+  QueueService.getStatus(props.queueId).then(setDetails).catch(handleApiErrors);
```

Comment:
This would make the the API call on all re-renders. We should enclose it in a `useEffect` hook, so that it gets called only on changes to `props.queueId`:
```
  useEffect(() => {
    async function fetchData() {
      const response = await QueueService.getStatus(props.queueId).catch(handleApiErrors);
      setQueueStatusResponse(response);
    }
    fetchData();
  }, [props.queueId]);
```

Reply:
https://www.robinwieruch.de/react-hooks-fetch-data

Reply:
You're right. Missed this.